File: js/reverse-integer.js
var reverse = function (x) {
  if (x < 0) return -reverse(-x)
  let n = +x.toString().split('').reverse().join('')
  // >>0 返回n的32位带符号整数
  if (n >> 0 !== n) return 0
  return n
}

File: py/reverse-integer.py
class Solution:
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        if x < 0:
          return self.reverse(x * -1) * -1
        n = str(x)[::-1]
        n = int(n)
        # 长度溢出的情况
        if n > 2**31 - 1:
           return 0
        
        return n
        